### PR TITLE
E2E tests for sink: TLS key pair rotation

### LINKED ISSUE #3375: Eventing TLS: E2E tests for Sink TLS key pair rotation
**Problem**
We should have rekt-based E2E tests that tests TLS key pairs rotation for sink.
- Part of #3221

We should make use of:
- https://github.com/knative-extensions/reconciler-test/pull/548
- https://github.com/knative-extensions/reconciler-test/pull/549

Each test flow would be:

Install target resources
- rotate certs (https://github.com/knative-extensions/reconciler-test/pull/549)
- verify new rotated certs are used (https://github.com/knative-extensions/reconciler-test/pull/548)

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)** 
Which persona is this feature for?

**Exit Criteria**
A measurable (binary) test that would indicate that the problem has been resolved.

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve?
1
**Additional context (optional)**
- related to https://github.com/knative/eventing/issues/7039

For Channel tests, this issue depends on:
- https://github.com/knative-extensions/eventing-kafka-broker/issues/3189

### PR DESCRIPTION
**Pull request by Leo6Leo**
Fixes #3375 

## Proposed Changes

-  Add the rekt test for keypair rotation for kafka sink

## Bugs Found

- [x] TLS port 8443 is not exposed for https-container

- [x] In the `control-plane/config/eventing-kafka-broker/100-sink/100-kafka-sink.yaml`, there is no custom resource definition for the addressable 

## Action Items

- [ ]  Backporting this ticket to upstream and midstream

### CODE REVIEW

File: test/rekt/features/sink_tls.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+	f.Setup("Rotate ingress certificate", certificate.Rotate(certificate.RotateCertificate{
+		Certificate: types.NamespacedName{
+			Namespace: system.Namespace(),
+			Name:      ingressCertificateName,
+		},
+	}))
+
+	f.Setup("Sink has HTTPS address", kafkasink.ValidateAddress(sink, addressable.AssertHTTPSAddress))
+	//
```
Comment (pierDipi) [inline test/rekt/features/sink_tls.go:67]:
```suggestion

```

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Comment by Cali0707** [comment]
@Leo6Leo can you cherry pick this manually to release-1.11?

**Comment by Leo6Leo** [comment]
@Cali0707 There are some issues when merging the broker TLS keypair rotation tickets. @pierDipi mentioned that he has a PR to be merged, and after that the problem will be resolved. Then I will be manually cherry-pick on this PR just to not dealing with the merging conflicts.

**Pull request merged**